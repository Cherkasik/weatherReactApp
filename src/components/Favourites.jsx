import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import FavTown from './FavTown';
import * as favouritesSelector from '../favourites/selectors';
import { getFavourites, addFavourite } from '../favourites/actions';

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 40px;
`;

const Text = styled.div`
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
`;

const NewCityInput = styled.input`
  border: none;
  background-color: #F1F1F1;
  color: #8B9396;
  padding: 0 12px;
  width: 400px;
  margin-right: 20px;
  border-radius: 2px;
  height: 100%;
`;

const AddButton = styled.button`
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  padding: 0 14px;
  font-size: 25pt;
`;

const Towns = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-top: 40px;
  flex-wrap: wrap;
`;

class Favourites extends React.Component {
  componentDidMount() {
    this.props.getFavourites();
  }

  handleAddClick (e) {
    e.preventDefault();
    if (e.target[0].value !== '') {
      this.props.addFavourite(e.target[0].value);
      e.target[0].value = '';
    }
  }

  render() {
    const { favourites } = this.props;
    return (
      <>
        <Header>
          <Text>Избранное</Text>
          <form style={{display: 'flex'}} onSubmit={e => this.handleAddClick(e)}>
            <NewCityInput placeholder="Добавить новый город"/>
            <AddButton type="submit">+</AddButton>
          </form>
        </Header>
        <Towns>
          {favourites && favourites.map(townName => (
            <FavTown
              key={townName}
              name={townName}
            />
          ))}
        </Towns>
      </>
    );
  }
}

const mapStateToProps = state => ({
  favourites: favouritesSelector.getFavourites(state),
});

const mapDispatchToProps = {
  getFavourites,
  addFavourite,
};

export default connect(mapStateToProps, mapDispatchToProps)(Favourites);
