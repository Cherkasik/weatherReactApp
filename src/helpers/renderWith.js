import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import reducer from '../store/rootReducer';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  }
}
