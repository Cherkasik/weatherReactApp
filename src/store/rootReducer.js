import { combineReducers } from 'redux';
import geolocation from '../geolocation/reducer';
import favourites from '../favourites/reducer';
import weather from '../weather/reducer';

const appReducer = combineReducers({
  geolocation,
  favourites,
  weather,
});

const rootReducer = (state, actionValue) => {
  return appReducer(state, actionValue);
};

export default rootReducer;