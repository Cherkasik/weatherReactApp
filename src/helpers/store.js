import thunkMiddleware from "redux-thunk";
import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../store/rootReducer";

const middlewares = [thunkMiddleware];

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middlewares.push(logger);
}

export const store = compose(applyMiddleware(...middlewares))(createStore)(rootReducer);