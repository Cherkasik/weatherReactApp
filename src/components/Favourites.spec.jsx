import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Favourites from './Favourites';
import { store } from '../helpers/store';

describe('Favourites component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Favourites />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});