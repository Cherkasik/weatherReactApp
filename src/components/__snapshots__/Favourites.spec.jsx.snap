// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites component should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 40px;
}

.c1 {
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
}

.c2 {
  border: none;
  background-color: #F1F1F1;
  color: #8B9396;
  padding: 0 12px;
  width: 400px;
  margin-right: 20px;
  border-radius: 2px;
  height: 100%;
}

.c3 {
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  padding: 0 14px;
  font-size: 25pt;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Избранное
  </div>
  <form
    style="display: flex;"
  >
    <input
      class="c2"
      placeholder="Добавить новый город"
    />
    <button
      class="c3"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;
